server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -111,7 +111,7 @@ public Aggregator build(
                     execution = ExecutionMode.fromString(executionHint);
                 }
                 // In some cases, using ordinals is just not supported: override it
-                if (valuesSource instanceof ValuesSource.Bytes.WithOrdinals == false) {
+                if (valuesSource.hasOrdinals() == false) {
```

Comment:
This is unrelated, but I saw a chance to replace an `instanceof` with a capability check, and I took it.